floris/simulation/turbine.py

```diff
@@ -429,9 +443,28 @@ def axial_induction(
     )
 
 
+def simple_mean(array, axis=0):
+    return np.mean(array, axis=axis)
+
+def cubic_mean(array, axis=0):
+    return np.cbrt(np.mean(array ** 3.0, axis=axis))
+
+def simple_cubature(array, cubature_weights, axis=0):
+    weights = cubature_weights.flatten()
+    weights = weights * len(weights) / np.sum(weights)
+    product = (array * weights[None, None, None, :, None])
+    return simple_mean(product, axis)
+
+def cubic_cubature(array, cubature_weights, axis=0):
+    weights = cubature_weights.flatten()
+    weights = weights * len(weights) / np.sum(weights)
+    return np.cbrt(np.mean((array**3.0 * weights[None, None, None, :, None]), axis=axis))
+
 def average_velocity(
     velocities: NDArrayFloat,
-    ix_filter: NDArrayFilter | Iterable[int] | None = None
+    ix_filter: NDArrayFilter | Iterable[int] | None = None,
+    method: str = "cubic-mean",
+    cubature_weights: np.array | None = None
```

Comment:
`NDArrayFloat` instead of `np.array` for consistency?

Reply:
@rafmudaf I've gone ahead and made this change, please revert if this was incorrect.

Reply:
I used `np.array` since this is specifically a 1D array and `NDArrayFloat` is n-dimensional array. We don't have a correct type for this instance, but I like the form that you or Paul used in [FlorisInterface.same_flow_at_points](https://github.com/NREL/floris/blob/develop/floris/tools/floris_interface.py#L942) for `1DArrayFloat`.

Lets leave it as you've changed it here, and consider `1DArrayFloat` and `5DArrayFloat` an option for future refinement.